packages/app-shell-ui/src/utils/processConfig.ts

```diff
@@ -0,0 +1,214 @@
+  type ProcessedProvider,
+  type ProcessedServiceConfig,
+  type ProcessedView,
+} from "@hitachivantara/app-shell-shared";
+
+// Generate a unique key for internal tracking
+const generateKey = (): string => {
+  return `${Date.now()}-${Math.round(1000 * Math.random())}`;
+};
+
+// Create a map of route -> conditions from views for menu inheritance
+const createRouteConditionsMap = (
+  views: ProcessedView[],
+): Map<string, ProcessedCondition[]> => {
+  const map = new Map<string, ProcessedCondition[]>();
+
+  const processView = (view: ProcessedView) => {
+    if (view.route && view.conditions) {
+      const normalizedRoute = view.route.replace(/^\//, "");
+      map.set(normalizedRoute, view.conditions);
+    }
+    if (view.views) {
+      for (const nestedView of view.views) {
+        processView(nestedView);
+      }
+    }
+  };
+
+  for (const view of views) {
+    processView(view);
+  }
+  return map;
+};
+
+// Merges both explicit menu conditions and inherited view conditions using AND logic (inheritance logic)
+const applyViewConditionsToMenus = (
+  menus: ProcessedMenu[],
+  routeConditionsMap: Map<string, ProcessedCondition[]>,
+): ProcessedMenu[] => {
+  return menus.map((menu) => {
+    let hasChanges = false;
+    let updatedMenu: ProcessedMenu = menu;
+
+    // If menu has a target, inherit conditions from matching view
+    if (menu.target) {
+      const normalizedTarget = menu.target.replace(/^\//, "");
+      const viewConditions = routeConditionsMap.get(normalizedTarget);
+
+      if (viewConditions) {
+        // Merge explicit menu conditions with inherited view conditions
+        // Reuse the same condition instances (same globalIndex)
+        const mergedConditions = [
+          ...(menu.conditions || []),
+          ...viewConditions,
+        ];
+
+        if (mergedConditions.length > 0) {
+          updatedMenu = { ...menu, conditions: mergedConditions };
+          hasChanges = true;
+        }
+      }
+    }
+
+    if (menu.submenus) {
+      const updatedSubmenus = applyViewConditionsToMenus(
+        menu.submenus,
+        routeConditionsMap,
+      );
+
+      if (updatedSubmenus !== menu.submenus) {
+        updatedMenu = hasChanges
+          ? { ...updatedMenu, submenus: updatedSubmenus }
+          : { ...menu, submenus: updatedSubmenus };
+        hasChanges = true;
+      }
+    }
+
+    return hasChanges ? updatedMenu : menu;
+  });
+};
+
+/**
+ * Process App Shell config into AppShellModel and adds key properties and globalIndex to conditions
+ * Returns AppShellModel with internal configuration logic
+ */
+export default function processConfig(
+  rawConfig: HvAppShellConfig,
+): AppShellModel {
+  let globalIndex = 0;
+  const allConditions: ProcessedCondition[] = [];
+
+  // Helper to add key and process conditions
+  const registerElement = <
+    T extends { conditions?: any[] },
+    P extends { key: string; conditions?: ProcessedCondition[] },
+  >(
+    element: T,
```

Comment:
Should clarify it is a config element somehow, e.g.`configElement`.

Also, in the code, variables should make it clear which is which, config or model, element or condition. For example:

```ts
const conditions = configElement.conditions?.map((condConfig) => {
   const condModel : ConditionModel = {`...
        ...condConfig
 ```